### ISSUE TITLE
.zcompdump files

### ISSUE THREAD

**Issue by yannickwurm** [wurmlab/oswitch#41]
there's a lot of these files in my home dir. Is that oswitch? and can I delete them? Can we avoid generating them?

﻿﻿
![screenshot 2015-12-10 00 10 32](https://cloud.githubusercontent.com/assets/150113/11703021/8286323e-9ed3-11e5-9802-e4c58705da6e.png)

**Comment by yeban** [comment]
No, it's zsh. `.zcompdump` is used by zsh completion system to speed up completion. It's safe to delete them, but they will be recreated unless you add `-D` to `compinit` line in your zsh config. In this case a lot of them are created because each docker container is a different host. If you will notice it's hostname that is suffixed to each `.zcompdump` file.

oswitch could potentially reduce the number of files created by supplying the same hostname for say all biolinux containers. But I don't think it's a good idea – it would be a source of confusion. oswitch creates new containers every time, but seeing 'biolinux' in the shell prompt would make you think you are inside the same container even in different terminals.

**Comment by yannickwurm** [comment]
Ok makes sense

Typos courtesy of my iPhone

On 10 Dec 2015, at 07:27, Anurag Priyam <notifications@github.com<mailto:notifications@github.com>> wrote:


No, it's zsh. .zcompdump is used by zsh completion system to speed up completion. It's safe to delete them, but they will be recreated unless you add -D to compinit line in your zsh config. In this case a lot of them are created because each docker container is a different host. If you will notice it's hostname that is suffixed to each .zcompdump file.

oswitch could potentially reduce the number of files created by supplying the same hostname for say all biolinux containers. But I don't think it's a good idea - it would be a source of confusion. oswitch creates new containers every time, but seeing 'biolinux' in the shell prompt would make you think you are inside the same container even in different terminals.

-

**Comment by 000panther** [comment]
Well, I resolved the issue by just deleting .zcompdump - seems to be only a cache file anyway. Thanks.

**Comment by s777s** [comment]
Are you from Russia, mr. Yeban?) 
https://translate.google.com/#ru/en/%D1%83%D0%B5%D0%B1%D0%B0%D0%BD
